### ISSUE TITLE
Mobx Observable not updating

### ISSUE THREAD

**Issue by sourceship** [mobxjs/mobx#1749]
Hello All, 

I have a mobx observable im creating to get a basic understanding of mobx. 

I have a store that looks like this: 
```
import { observable, autorun, action } from "mobx";

class DataStore {
  @observable SelectedCustomer = "All Customers";
}
const dataStore = new DataStore();
export default dataStore;
```

and a view that looks like this: 
```
import PropTypes from "prop-types";
import React, { Component, StyleSheet, FlatList } from "react";
import dataStore from "../store/DataStore";
import { observer } from "mobx-react";


@observer
export default class DrawerMenu extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      customerDropdownList: []
    };
  }
  render() {
    return (
      <Container>


  <Text>{dataStore.SelectedCustomer}</Text>

      </Container>
    );
  }
}

````

My problem is that I cannot get to show it up. If I change my datastore to;
```
 SelectedCustomer = observable.box("All Customers");
```
and my view to:
```
  <Text>{dataStore.SelectedCustomer.view}</Text>

```
it will show up. but it will not update the view when i change the observable  value like here:
```
    dataStore.SelectedCustomer = text;

```

**Can you guys give me some direction on what i'm doing wrong and how I should be doing this instead?** 
I'm kinda lost.

**Comment by sourceship** [comment]
That makes sense. I am going to change my code and try that out, and get back to you. 

Oh and that was a typo, I mean to say, dataStore.SelectedCustomer.value
This also explains, why I had to say .value

Thank You so much for your help. I've been banging my head against this for quite some time.

**Comment by y4ure** [comment]
> JS primitives(string, number, boolean, null, undefined) functions are not observable by default, observable() function automatically converts an object, an array, or a map into an observable entity. 

@_@ This is strange as I use @observable on primitives in many places and have no problems updating or rendering. see example below:
https://codesandbox.io/s/oq857m2jy6

**Comment by sourceship** [comment]
```
{
  "name": "XXXXX",
  "version": "0.0.1",
  "private": true,
  "scripts": {
    "start": "node node_modules/react-native/local-cli/cli.js start",
    "test": "jest"
  },
  "dependencies": {
    "@around25/jwt-utils": "^1.0.1",
    "es6-symbol": "^3.1.1",
    "mobx": "^4.3.1",
    "mobx-react": "^5.1.0",
    "native-base": "^2.8.0",
    "react": "16.3.1",
    "react-native": "0.55.4",
    "react-native-animatable": "^1.3.0",
    "react-native-awesome-alerts": "^1.1.1",
    "react-native-easy-grid": "^0.2.0",
    "react-native-elements": "^1.0.0-beta",
    "react-native-indicators": "^0.13.0",
    "react-native-keyboard-aware-scroll-view": "^0.7.2",
    "react-native-maps": "^0.21.0",
    "react-native-material-dropdown": "^0.11.1",
    "react-native-material-ui": "^1.30.1",
    "react-native-modal-dropdown": "^0.6.2",
    "react-native-simple-table": "0.0.3",
    "react-native-vector-icons": "^5.0.0",
    "react-navigation": "^2.0.1",
    "react-navigation-material-bottom-tabs": "^0.1.2"
  },
  "devDependencies": {
    "babel-jest": "22.4.3",
    "babel-plugin-transform-decorators-legacy": "^1.3.5",
    "babel-preset-react-native": "4.0.0",
    "jest": "22.4.3",
    "jsc-android": "^224109.1.0",
    "react-test-renderer": "16.3.1"
  },
  "jest": {
    "preset": "react-native"
  },
  "babel": {
    "plugins": [
      "transform-decorators-legacy"
    ]
  }
}

```

I dont think so. That is my package.json^. One thing though is I dont have a .bablerc file, I have it inside my package.json at the bottom; not sure if that has anything to do with it.

**Comment by wellrus** [comment]
May be you use chrome ver 69?

**Comment by mweststrate** [comment]
@y4ure primitives are never observable. Object properties however can be (that is what `@observable` does), it is a slightly different thing ;)

@sourceship your original code looks fine to me, could you setup a test repo, or reproduce it in a sandbox? To check whether this is a setup issue, consider using `decorate(DataStore, { SelectedCustomer: observable })` rather then the decorate syntax. There have been some issues with babel on RN lately, depending on the exact version the default babel setup might use some beta version of babel that gives issues

**Comment by sourceship** [comment]
Thank You for your help. 

Yes I can, sorry for the delay, I get back in the office on Monday and can submit a Test Repo. 

Will post an update soon.

**Comment by mweststrate** [comment]
@sourceship had no time to look earlier. Are you still stuck on this one? FWIW mobx usage and project setup looks correct

**Comment by sourceship** [comment]
@mweststrate  Hey its no biggie,

 I gave it another go a few days ago and kept facing the same problem. I ended up getting desperate and trying out a different state management repo, react-easy-state and had success with that. 

I'm not doing complex state management; I have a collection of objects that I build my view off of, and update my view when that collection changes, mobx may have been overkill for my project. 

Not quite sure why mobx wasnt working in my scenario, it did not work in the example repo I posted, but I react-easy-state ended up accomplishing what I was looking for. 

Thanks for your help though, I really appreciate your time!
